src/interface/metadata.cpp

```diff
@@ -141,6 +141,13 @@ std::array<int, 6> Metadata::GetArrayDims(std::weak_ptr<MeshBlock> wpmb,
       arrDims[i + 3] = shape[i];
     for (int i = N; i < 3; i++)
       arrDims[i + 3] = 1;
+  } else if (Where() == Particle) {
+    assert(N >= 1 && N <= 5);
+    arrDims[0] = 1; // To be updated by swarm based on pool size before allocation
+    for (int i = 0; i < N; i++)
+      arrDims[i + 1] = shape[i];
+    for (int i = N; i < 5; i++)
+      arrDims[i + 1] = 1;
```

Comment:
I don't fully understand this change. Why do you need to specify `Particle` in the metadata if you're adding it to a swarm? Is it because this logic lives in the metadata?

If so, maybe we could remove the need to `AddSwarmValue` and just `Add` but with particle metadata, which short-circuits to adding a swarm?

Reply:
Somehow inside `Metadata` we need to know if we're dealing with a `Particle` variable because these variables have a specific `arrDims` (first index is the index inside the swarm pool, the other five indices are for up to a 5D dataset per particle). 

If we remove `AddSwarmValue` would that cause issues for having multiple swarms in one application? Because `AddSwarmValue` adds the `ParticleVariable` to a specific `Swarm`

Reply:
Yes adding `Metadata::Particle` inside `AddSwarmValue()` is a good idea, done